letsencrypt-auto-source/letsencrypt-auto

```diff
@@ -649,14 +652,8 @@ Bootstrap() {
   fi
 }
 
-TempDir() {
-  mktemp -d 2>/dev/null || mktemp -d -t 'le'  # Linux || macOS
-}
-
-
-
 if [ "$1" = "--le-auto-phase2" ]; then
-  # Phase 2: Create venv, install LE, and run.
+    # Phase 2: Create venv, install LE, and run.
```

Comment:
So it's not a big deal, but I think it'd be nice if we could handle indentation a little more gracefully. Previously, the only files that were included by the template were definitions of shell script functions or contents of files to be written to disk (or the version number). This is the first time that the template system is being used to include blocks of code that should be indented relative to the template. As a side effect of the way it currently works and the way the code was initially written, we have slightly off indentation like this and the need to indent the majority of entire files like we see in the `phase-?/main.sh` files.

I'm open to any ideas you have to solve this, but the current approach I have in mind is to stick with the previous approach of when including shell script code, only include function definitions. The way this would work is the current places in `letsencrypt-auto.template` that these files are included become a call to the function defined in that file rather than something to be expanded by `build.py`. Then the `main.sh` files would become standalone functions that could be included by the template anywhere before they are used.

Hopefully this makes sense and let me know if you have any better ideas or questions about what I'm proposing here!